storage/netstore.go

```diff
@@ -99,7 +99,7 @@ type NetStore struct {
 	requestGroup     singleflight.Group
 	RemoteGet        RemoteGetFunc
 	logger           log.Logger
-	recoveryCallback func(ctx context.Context, chunkAddress chunk.Address) error // this is the function callback when a chunk failed to be retrieved
+	recoveryCallback func(ctx context.Context, chunkAddress chunk.Address, publisher string) error // this is the callback to be executed when a chunk fails to be retrieved
```

Comment:
`publisher` _may_ not be a param here, but rather extracted from `ctxt`

Reply:
it makes sense to extract from context within the recovery function

Reply:
removed this param, expecting code to be pushed to extract this data from `ctx`